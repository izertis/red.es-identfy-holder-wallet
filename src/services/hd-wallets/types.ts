import { derivationPathFormats } from './constants'

type randomType = 'random'

export type settingType = {
	m?: undefined
	Z?: string
	R?: string | randomType
	SSSSS?: [string, string, string, string, string] | randomType
	W?: string | randomType
	M?: string
	T?: string
	N?: string
	B?: randomType
	BB?: [string, string] | randomType
}
export type validKeysType = keyof settingType

export type derivationPathFormat =
	| derivationPathFormats.MAIN_IDENTITY
	| derivationPathFormats.NETWORK_DID
	| derivationPathFormats.INTERACTING_DID
