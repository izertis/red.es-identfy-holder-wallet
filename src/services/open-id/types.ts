import { Jwt, JwtHeader, JwtPayload } from 'jsonwebtoken'
import { ModalPropsWithoutIsVisible } from '../../context/Modal.context'

export type ModalType = {
	showModal?: (props?: ModalPropsWithoutIsVisible) => void
	t?: (s: string) => string
	showMessage?: (x: { content: string; type: string }) => void
}

export type Grants = {
	authorization_code: {
		issuer_state: string
	}
	'urn:ietf:params:oauth:grant-type:pre-authorized_code'?: {
		'pre-authorized_code': string
		user_pin_required: boolean
	}
}

export type PreAuthorizeResponse = {
	access_token: string
	token_type: string
	expires_in: number
	id_token: string
	c_nonce: string
	c_nonce_expires_in: number
}

export type CredentialOffer = {
	credential_issuer: string
	credentials: {
		format: string
		types: string[]
		trust_framework: {
			name: string
			type: string
			uri: string
		}
	}[]
	grants: Grants
}

export type PresentationOffer = {
	aud: string
	client_id: string
	exp: number
	iat: number
	iss: string
	nonce: string
	presentation_definition: PresentationDefinition
	redirect_uri: string
	response_mode: string
	response_type: string
	scope: string
	state?: string
}

export type OpenIdCredentialIssuer = {
	credential_issuer: string
	authorization_server: string
	credential_endpoint: string
	deferred_credential_endpoint: string
	credentials_supported: {
		format: string
		types: string[]
		trust_framework: {
			name: string
			type: string
			uri: string
		}
		display: {
			name: string
			locale: string
		}[]
	}[]
}

export type OpenIdConfiguration = {
	redirect_uris: string[]
	issuer: string
	authorization_endpoint: string
	token_endpoint: string
	jwks_uri: string
	scopes_supported: string[]
	response_types_supported: string[]
	response_modes_supported: string[]
	grant_types_supported: string[]
	subject_types_supported: string[]
	id_token_signing_alg_values_supported: string[]
	request_object_signing_alg_values_supported: string[]
	request_parameter_supported: boolean
	request_uri_parameter_supported: boolean
	token_endpoint_auth_methods_supported: string[]
	request_authentication_methods_supported: {
		[key: string]: string[]
	}
	vp_formats_supported: {
		[key: string]: {
			alg_values_supported: string[]
		}
	}
	subject_syntax_types_supported: string[]
	subject_syntax_types_discriminations: string[]
	subject_trust_frameworks_supported: string[]
	id_token_types_supported: string[]
}

export type CredentialResponse = {
	format: 'jwt_vc'
	credential: string
	acceptance_token: string
}

export type AuthorizationRequest = {
	response_type: 'code'
	client_id: string
	redirect_uri: string
	scope: string
	issuer_state?: string
	state?: string
	authorization_details: string
	nonce?: string
	code_challenge?: string
	code_challenge_method?: 'S256' | 'plain'
	client_metadata?: string
}

export type HolderServiceWalletMetadata = {
	authorization_endpoint: string
	response_types_supported: string[]
	vp_formats_supported: {
		jwt_vp: {
			alg_values_supported: string[]
		}
		jwt_vc: {
			alg_values_supported: string[]
		}
	}
	scopes_supported: string[]
	subject_types_supported: string[]
	id_token_signing_alg_values_supported: string[]
	request_object_signing_alg_values_supported: string[]
	subject_syntax_types_supported: string[]
	id_token_types_supported: string[]
}

export type AuthorizationDetail = {
	type: 'openid_credential'
	locations?: string[]
	format: 'jwt_vp' | 'jwt_vc' | string
	types: string[]
}

export type AuthorizationResponse = {
	code: string
	state?: string
}

export type GrantType = 'urn:ietf:params:oauth:grant-type:pre-authorized_code' | 'preauthorized_code'

export type TokenRequest = {
	redirect_uri: string
	grant_type: GrantType
	client_id?: string
	code?: string
	client_assertion_type?: boolean
	code_verifier?: string
	user_pin?: string
	'pre-authorized_code'?: string
}

export type TokenResponse = {
	access_token: string
	refresh_token: string
	token_type: string
	expires_in: number
	id_token: string
	c_nonce: string
	c_nonce_expires_in: number
}

export type VPTokenRequestPayload = {
	iss: string
	aud: string
	exp: number
	response_type: string
	response_mode: string
	client_id: string
	redirect_uri: string
	scope: string
	nonce: string
	presentation_definition_uri: string
}

export type AuthorizationRequestResponse = {
	client_id: string
	key: string
	nonce: string
	redirect_uri: string
	request_uri: string
	response_mode: string
	response_type: string
	scope: string
	presentation_definition: PresentationDefinition
	presentation_definition_uri: string
	request: string
	error: string
	error_description: string
}

export type DeferredCredential = {
	credential_endpoint: string
	acceptance_token: string
}

export type PresentationDefinition = {
	id: string
	format: Format
	input_descriptors: InputDescriptor[]
	purpose?: string
}

type InputDescriptor = {
	id: string
	name?: string
	format?: Format
	constraints?: {
		fields: FieldConstraint[]
	}
}

type FieldConstraint = {
	path: string[]
	filter: {
		type: string
		contains: {
			const: string
		}
	}
}

export type Format = {
	jwt_vc?: {
		alg: string[]
	}
	jwt_vp?: {
		alg: string[]
	}
	jwt_vc_json?: {
		alg: string[]
	}
	jwt_vp_json?: {
		alg: string[]
	}
}

export type TrustedFramework = 'lacchain' | 'alastria' | 'ebsi'

export type CredentialData = {
	id: string
	issuer: string
	validFrom: any
	expirationDate: any
	timestamp?: Date
	credential: Credential
	status: 'Active' | 'Revoked'
}

// CREDENTIALS SCHEMA
export type Vc = {
	'@context': string[]
	id: string
	type: string[]
	credentialSchema: {
		id: string
		type: string
	}
	credentialSubject: {
		id: string
	}
	validFrom?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	validUntil?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	expirationDate?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	issuanceDate: string
	issuer: string
	issued?: string
	credentialStatus?: W3CCredentialStatus
	proof?: EmbeddedProof
	[x: string]: any
}

export interface EmbeddedProof {
	id?: string
	type: string
	proofPurpose: string
	verificationMethod: string
	created?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	expires?: string // Date timestamp. Example: "2010-01-01T19:23:24Z",
	domain?: string
	challenge?: string
	proofValue: string
	previousProof?: string | string[]
	nonce?: string
}

export interface W3CCredentialStatus {
	id?: string
	type: string
	[key: string]: any
}

export interface StatusListEntry2021 {
	id: string
	type: 'StatusList2021Entry'
	statusPurpose: 'revocation' | 'suspension'
	statusListIndex: string
	statusListCredential: string
}

export interface StatusListCredentialData {
	id: string
	type: 'StatusList2021'
	statusPurpose: 'revocation' | 'suspension'
	encodedList: string
}

export type Credential = {
	header: JwtHeader
	payload: JwtPayload & {
		vc: Vc
	}
}

export type chosenCredential = {
	format: 'jwt_vc' | 'ldp_vc'
	trust_framework: { name: string; type: string; uri: string }
	types: ['VerifiableCredential', 'VerifiableAttestation', string]
}

export type PresentationData = {
	id: string
	issuer: string
	validFrom: any
	expirationDate: any
	timestamp: Date
	credential: Credential
	status: 'Active' | 'Revoked'
}

export type ActionLog = {
	id: string
	actionType: 'Credential added' | 'Credential expired' | 'Presentation emitted' | 'Presentation revoked'
	issuer: string
	date: Date | string
	data?: any
}
