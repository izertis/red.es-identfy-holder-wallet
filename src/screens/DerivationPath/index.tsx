import React, { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Card } from 'react-native-paper'
import { RootStackScreenProps } from '../../../types'
import Button from '../../components/atomic_components/Button'
import PressToCopy from '../../components/PressToCopy'
import { SCREEN } from '../../constants/screens'
import DerivationPathI18nKeys from './i18n/keys'
import DerivationPathStyled from './styles'

const DerivationPath = ({ route, navigation }: RootStackScreenProps<'DerivationPath'>) => {
	const { t } = useTranslation(SCREEN.DerivationPath)

	const { derivationPath }: any = route.params

	const [isSeedBlur, setIsSeedBlur] = useState<boolean>(true)

	return (
		<DerivationPathStyled.MainContainer>
			<DerivationPathStyled.BackButton onPress={() => navigation.pop()}>
				{t(DerivationPathI18nKeys.GO_BACK)}
			</DerivationPathStyled.BackButton>
			<DerivationPathStyled.Container>
				<DerivationPathStyled.Title testID='title'>
					{t(DerivationPathI18nKeys.TITLE)}
				</DerivationPathStyled.Title>
				<DerivationPathStyled.Subtitle>{t(DerivationPathI18nKeys.SUBTITLE)}</DerivationPathStyled.Subtitle>

				{isSeedBlur ? (
					<DerivationPathStyled.PhraseContainer onPress={() => setIsSeedBlur(!isSeedBlur)}>
						<DerivationPathStyled.MainCard justifyContent='flex-end' alignItems='center'>
							<Card.Content>
								<DerivationPathStyled.HiddenTitle>
									{t(DerivationPathI18nKeys.HIDDEN_TITLE)}
								</DerivationPathStyled.HiddenTitle>
							</Card.Content>
						</DerivationPathStyled.MainCard>
					</DerivationPathStyled.PhraseContainer>
				) : (
					<DerivationPathStyled.PhraseContainer>
						<PressToCopy onPress={() => setIsSeedBlur(!isSeedBlur)} value={derivationPath}>
							<DerivationPathStyled.MainCard>
								<Card.Content>
									<DerivationPathStyled.Content>{derivationPath}</DerivationPathStyled.Content>
								</Card.Content>
							</DerivationPathStyled.MainCard>
						</PressToCopy>
					</DerivationPathStyled.PhraseContainer>
				)}
				{!isSeedBlur && (
					<DerivationPathStyled.CopyText>
						{t(DerivationPathI18nKeys.LONG_PRESS)}
					</DerivationPathStyled.CopyText>
				)}

				<DerivationPathStyled.ContainerBottom>
					<Button
						onPress={() => {
							navigation.navigate('NetworkAuth')
						}}
					>
						{t(DerivationPathI18nKeys.BUTTON)}
					</Button>
				</DerivationPathStyled.ContainerBottom>
			</DerivationPathStyled.Container>
		</DerivationPathStyled.MainContainer>
	)
}

export default DerivationPath
