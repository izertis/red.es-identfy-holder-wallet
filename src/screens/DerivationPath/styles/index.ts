import defaultStyles from "./defaultStyles"
import proyectSyles from "./proyectStyles"

const DerivationPathStyled = {
  ...defaultStyles,
  ...proyectSyles
}

export default DerivationPathStyled
