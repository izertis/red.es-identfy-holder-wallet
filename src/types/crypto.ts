export type JWT = `ey${string}.ey${string}.${string}`