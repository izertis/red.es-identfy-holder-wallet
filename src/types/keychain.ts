import { ActionLog, CredentialData, DeferredCredential, PresentationData } from '../services/open-id/types'

export type KeyChainData = {
	encryptedPrivateKey?: string | ArrayBuffer
	did?: { lacchain?: string; alastria?: string; ebsi?: string }
	JWKKeys?: { privateKeyJWK: string; publicKeyJWK: string; privateKeyHex: string; publicKeyHex: string }
	signatureKeyPair?: { publicKey: string; privateKey: string }
	credentials?: CredentialData[]
	presentations?: PresentationData[]
	deferred_credentials?: DeferredCredential[]
	actionLogs?: ActionLog[]
}
