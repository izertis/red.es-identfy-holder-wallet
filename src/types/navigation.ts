import { MaterialTopTabScreenProps } from '@react-navigation/material-top-tabs'
import { NativeStackNavigationOptions } from '@react-navigation/native-stack'

export type navigationScreensConfigType = {
	name: string
	component: (props: any) => JSX.Element
	options:
		| NativeStackNavigationOptions
		| MaterialTopTabScreenProps<{}>
		| ((x: any) => NativeStackNavigationOptions | MaterialTopTabScreenProps<{}>)
}
