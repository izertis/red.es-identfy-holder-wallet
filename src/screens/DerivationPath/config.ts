import i18next from '../../../i18n.config'
import { SCREEN } from "../../constants/screens"
import { navigationScreensConfigType } from "../../types/navigation"
import es from "./i18n/es"
import DerivationPath from '.'

const ScreenName = SCREEN.DerivationPath

i18next.addResourceBundle("es", ScreenName, es)

const DerivationPathConfig: navigationScreensConfigType = {
  name: ScreenName,
  component: DerivationPath,
  options: { headerShown: false },
}

export default DerivationPathConfig
