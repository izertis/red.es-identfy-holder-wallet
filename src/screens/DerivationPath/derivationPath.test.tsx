import { RouteProp } from '@react-navigation/native'
import { NativeStackNavigationProp } from '@react-navigation/native-stack'
import { render } from '@testing-library/react-native'
import DerivationPath from '.'
import { RootStackParamList } from '../../../types'

// react-native-clipboard jest-mock
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')

type ScreenNavigationProp = NativeStackNavigationProp<RootStackParamList, 'DerivationPath'>

type DerivationPathRouteProp = RouteProp<RootStackParamList, 'DerivationPath'>

const mockedNavigation: ScreenNavigationProp | any = {
	navigate: jest.fn(),
}

const mockedRoute: DerivationPathRouteProp = {
	key: 'mocked-key',
	name: 'DerivationPath',
	params: {
		derivationPath: 'mocked-derivation-path',
	},
}

describe('DerivationPath screen', () => {
	it('should render DerivationPath screen', () => {
		const result = render(<DerivationPath navigation={mockedNavigation} route={mockedRoute} />)

		expect(result.toJSON()).toMatchSnapshot()
	})
	it('Should renders items correctly', () => {
		const { getAllByRole, getByTestId } = render(
			<DerivationPath navigation={mockedNavigation} route={mockedRoute} />
		)

		const title = getByTestId('title')
		const continueBtn = getAllByRole('button')

		expect(title).toBeDefined()
		expect(continueBtn).toBeDefined()
	})
})
