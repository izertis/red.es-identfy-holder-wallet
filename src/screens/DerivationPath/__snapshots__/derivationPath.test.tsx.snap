// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DerivationPath screen should render DerivationPath screen 1`] = `
<View
  className="StyledNativeComponent sc-eDvSVe cGYWIG"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "transparent",
        "borderRadius": 20,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="button-container-outer-layer"
  >
    <View
      className="sc-jSUZER hxcjZp"
      collapsable={false}
      style={
        {
          "backgroundColor": "transparent",
          "borderColor": "transparent",
          "borderRadius": 20,
          "borderStyle": "solid",
          "borderWidth": 0,
          "flex": undefined,
          "minWidth": 64,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="button-container"
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            {
              "borderRadius": 20,
            },
          ]
        }
        testID="button"
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "marginLeft": 12,
                  "marginRight": -4,
                },
                {
                  "marginLeft": 16,
                  "marginRight": -16,
                },
                {
                  "marginLeft": 12,
                  "marginRight": -8,
                },
              ]
            }
            testID="button-icon-container"
          >
            <Text />
          </View>
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "writingDirection": "ltr",
                },
                [
                  {
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  [
                    {
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    false,
                    {
                      "marginHorizontal": 16,
                    },
                    undefined,
                    false,
                    {
                      "color": "rgba(103, 80, 164, 1)",
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    {
                      "color": "#eb5b5b",
                    },
                  ],
                ],
              ]
            }
            testID="button-text"
          >
            GO_BACK
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    className="sc-kDvujY dcxJXj"
  >
    <Text
      bold={true}
      className="sc-bcXHqe sc-dkrFOg sc-gKPRtg hXdTLw fuygvB kpWXDE"
      fontFamily="Montserrat-Bold"
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
      testID="title"
    >
      TITLE
    </Text>
    <Text
      className="sc-bcXHqe sc-gswNZR sc-iBYQkv laZgtI kEJapR jecNku"
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      SUBTITLE
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "rgba(247, 243, 249, 1)",
            "borderRadius": 12,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.15,
            "shadowRadius": 3,
          }
        }
        testID="card-container-outer-layer"
      >
        <View
          alignItems="center"
          className="sc-csuSiG jiIJTx"
          collapsable={false}
          justifyContent="flex-end"
          style={
            {
              "backgroundColor": "rgba(247, 243, 249, 1)",
              "borderRadius": 12,
              "elevation": 1,
              "flex": undefined,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            }
          }
          testID="card-container"
        >
          <View
            style={
              [
                {
                  "flexShrink": 1,
                },
                undefined,
              ]
            }
            testID="card"
          >
            <View
              style={
                [
                  {
                    "paddingHorizontal": 16,
                  },
                  {
                    "paddingVertical": 16,
                  },
                  undefined,
                ]
              }
            >
              <Text
                bold={true}
                className="sc-bcXHqe sc-dkrFOg sc-bqWxrE hXdTLw fuygvB eVCsWZ"
                fontFamily="Montserrat-Bold"
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {
                      "writingDirection": "ltr",
                    },
                    undefined,
                  ]
                }
              >
                HIDDEN_TITLE
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      className="sc-jrcTuL bXAgwe"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  {
                    "backgroundColor": "#eb5b5b",
                    "borderRadius": 15,
                    "fontFamily": "Muli",
                    "height": 39,
                    "width": 224,
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 12,
                        },
                        undefined,
                        false,
                        {
                          "color": "rgba(103, 80, 164, 1)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        {
                          "color": "#000",
                          "fontFamily": "Muli",
                          "letterSpacing": 0.57,
                        },
                      ],
                    ],
                  ]
                }
                testID="button-text"
              >
                BUTTON
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
